src/game_battle.cpp

```diff
@@ -81,6 +81,12 @@ void Game_Battle::Init() {
 		return false;
 	});
 
+	if (Game_Temp::battle_check_surprise_attack) {
+		Game_Temp::battle_first_strike = Utils::GetRandomNumber(1, 100) <=
+			(Main_Data::game_party->GetAverageAgility() > Main_Data::game_enemyparty->GetAverageAgility() ? 5 : 3);
+		//TODO: IMPLEMENT 2k3 DIFFERENT SURPRISE MODES
```

Comment:
@CherryDT

Do you agree with 5% and 3% here?

Reply:
Also maybe we should just move this logic to `Game_Map` when random encounters are triggered and set `battle_first_strike` there accordingly. This would avoid adding yet another `Game_Temp` global.

Reply:
I spent a lot of time to try and figure out how all these battle types behave... this is the result: https://gist.github.com/CherryDT/8984e216dca3c0cd3c84f80b57006c16

Regarding your change though - I didn't find anything related to that, and I'm a bit unsure what it is doing...